lib/active_merchant/billing/gateways/firstdata_e4.rb

```diff
@@ -263,7 +264,7 @@ def commit(action, request, credit_card = nil)
 
         Response.new(successful?(response), message_from(response), response,
           :test => test?,
-          :authorization => response_authorization(action, response, credit_card),
+          :authorization => successful?(response) ? response_authorization(action, response, credit_card) : nil,
```

Comment:
I've updated the fallback value to be an empty string instead of nil for consistency.